core/API/CORSHandler.php

```diff
@@ -18,40 +20,85 @@ class CORSHandler
      */
     protected $domains;
 
+    private $logger;
+
     public function __construct()
     {
         $this->domains = Url::getCorsHostsFromConfig();
+        $this->logger = StaticContainer::get(LoggerInterface::class);
+
     }
 
+    /**
+     * This method is set header for the request to pass browser cross domain checks.
+     * By default, it allow from all hosts
+     * @throws \Exception
```

Comment:
Might also be good to add a small note, that this method might make a hard `exit;` for preflight requests.